e2e_samples/temperature_events/LoadTesting/azure-pipeline.yml

```diff
@@ -5,33 +5,39 @@ trigger:
 pool:
   vmImage: ubuntu-latest
 
+# Store your eventhub connection string, eventhub name, and subscription id in keyvault.
+# In Keyvault:
+# TemperatureDevice-name: name of the eventhub.
+# TemperatureDevice-conn: connection string of the eventhub.
+# subscription-id: subscription Id of the project.
+
+# Then, create a variable group to get those secrets from keyvault.
 variables:
-  DeviceCount: 10
-  MessageCount: 1000
-  Interval: 100
-  ContainerCount: 2
-  ResourceGroup: 'dataops-temperature-events'
-  EvhNamespace: 'evh-temperature-test'
-  EvhName: 'temperature-processing'
-  ServiceConnection: dataops-temperature-events-sn
-  KeyVaultName: 'kv-temperature-test'
+- group: load-testing-secrets
+- name: DeviceCount
```

Comment:
Should we just add these values in variable group as well?

Reply:
Some are environment level that are dynamic in the Terraform.  Others are just specific knobs to tweak for the load test.  There are 7 variables left, of them:

- DeviceCount - Leave as pipeline variable
- MessageCount - Leave as pipeline variable
- Interval - Leave as pipeline variable
- ContainerCount - Leave as pipeline variable
- ResourceGroup - **Could move to variable group?** ✔
- ServiceConnection - Must be a pipeline variable ❌
- KeyVaultName - Must be a pipeline variable ❌

Reply:
thanks for catching the keyvaultname var! let me remove it.

btw, does service connection has to be a pipeline variable(in yaml)? I think defining it in variable group works as well :D